src/helpers.js

```diff
@@ -819,7 +819,7 @@ export function getNodeDevSource(dev) {
 
         source = `${bus}.${device}`;
     } else {
-        throw new Error(`getNodeDevSource: unsupport device type '${dev.type}'`);
+        console.warn(`getNodeDevSource: unsupport device type '${dev.type}'`);
```

Comment:
Fair enough, but then you actually need to handle this in the caller:
```
    devices.forEach(dev => {
        args.push("--add-device", "--hostdev", getNodeDevSource(dev));
    });
```

This is going to crash on an invalid command line (or even data type).

I suppose the following commit tries to avoid that by restricting the supported device types -- but code should be more defensive, as the lookup could fail for usb/pci as well.

Reply:
So I fixed it up. We use those 2 functions for 2 use cases basically, and here is what can happen in both situations:
1. Show device's source to in the UI. This is not critical, so I think here it's sufficient that console.warn() just prints the message and that's it.
2. Use device's source for attach/detach operations. This is critical, and should fail with a "throw" error. This is in both cases caught buy standard .then().catch(), and then dealth with by a caller (e.g. showing the error in the modal),